src/net_processing.cpp

```diff
@@ -1661,7 +1661,6 @@ bool PeerManagerImpl::HasDesirableServiceFlags(ServiceFlags services)
 ServiceFlags PeerManagerImpl::GetDesirableServiceFlags(ServiceFlags services)
 {
     if (services & NODE_NETWORK_LIMITED) {
-        LOCK(::cs_main);
         if (m_initial_sync_finished) return ServiceFlags(NODE_NETWORK_LIMITED | NODE_WITNESS);
```

Comment:
Is relaxed ordering ok for this? We don't need to call `m_initial_sync_finished.load()`?

Reply:
Why relaxed? look at the [operator](https://en.cppreference.com/w/cpp/atomic/atomic/operator_T). Is equivalent to `load`, which uses [std::memory_order_seq_cst](http://en.cppreference.com/w/cpp/atomic/memory_order).